src/libmain.c

```diff
@@ -141,10 +141,41 @@ static GOptionEntry entries[] =
 	{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose_mode, N_("Be verbose"), NULL },
 	{ "version", 'V', 0, G_OPTION_ARG_NONE, &show_version, N_("Show version and exit"), NULL },
 	{ "dummy", 0, G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_NONE, &dummy, NULL, NULL }, /* for +NNN line number arguments */
+	/* add new options here *and* in `optentries_aux' below */
 	{ NULL, 0, 0, 0, NULL, NULL, NULL }
 };
 
 
+GeanyOptionEntryAux optentries_aux[] = {
+	{FALSE}, /* "column" */
+	{TRUE},  /* "config" */
+	{FALSE}, /* "ft-names */
+	{FALSE}, /* "generate-tags" */
+	{FALSE}, /* "no-preprocessing" */
+#ifdef HAVE_SOCKET
+	{FALSE}, /* "new-instance" */
+	{TRUE},  /* "socket-file" */
```

Comment:
FALSE = "when reproducing the command line options passed to this instance to launch a New Window, ignore this option".

Since "-i" is automatically appended in utils_start_new_geany_instance(), it doesn't matter whether it's passed to this (launching) instance.